code/object/objcollide.cpp

```diff
@@ -1042,3 +1042,28 @@ void obj_sort_and_collide(SCP_vector<int>* Collision_list)
 	}
 	obj_find_overlap_colliders(sort_list_y, sort_list_z, 2, true);
 }
+
+void collide_apply_gravity_flags_weapons() {
+	for (object* obj = GET_FIRST(&obj_used_list); obj != END_OF_LIST(&obj_used_list); obj = GET_NEXT(obj)) {
+		if (obj->type != OBJ_WEAPON)
```

Comment:
you'll want to skip objects with the should-be-dead flag

(I don't think it matters here, but just for consistency)